app/assets/javascripts/angular/services/annotation_refresher.js.erb

```diff
@@ -1,31 +1,73 @@
-angular.module("Prometheus.services").factory('AnnotationRefresher', ["$http", "VariableInterpolator", function($http, VariableInterpolator) {
-  return function(graph, scope) {
-    var tags = graph.tags.map(function(e) {
-      if (e.name) {
-        var n = VariableInterpolator(e.name, scope.vars);
-        return n.split(",").map(function(s) { return s.trim(); });
-      } else {
-        return "";
+angular.module("Prometheus.services").factory('AnnotationRefresher', ["$http", "$q", "$timeout", "VariableInterpolator", function($http, $q, $timeout, VariableInterpolator) {
+  var MINUTE = 1000*60;
+  var promises = {};
```

Comment:
Yeah, I'm never cleaning it out. I could put them all on individual 1 min cooldowns? It may be an incorrect assumption, but I'm guessing that any long running dashboards see very little churn in their tags or endtime/range, so I just assumed the `now` would be the only case in which data would be getting refreshed.

guess there aren't any downsides to making the request every minute.